cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UnwindTest.scala

```diff
@@ -28,26 +28,76 @@ class UnwindTest extends DocumentingTestBase {
+      assertions = (p) => assertEquals(List(Map("x" -> 1, "y" -> "val"), Map("x" -> 2, "y" -> "val"), Map("x" -> 3, "y" -> "val"), Map("x" -> null, "y" -> "val")), p.toList)
     )
   }
   @Test def distinct_list() {
     testQuery(
-      title = "Create a distinct list",
+      title = "Creating a distinct list",
       text = "We want to transform a list of duplicates into a set using `DISTINCT`.",
       queryText = """WITH [1, 1, 2, 2] AS coll UNWIND coll AS x WITH DISTINCT x RETURN collect(x) AS set""",
       optionalResultExplanation = "Each value of the original list is unwound and passed through `DISTINCT` to create a unique set.",
       assertions = (p) => assertEquals(List(List(1,2)), p.columnAs[Int]("set").toList)
     )
   }
 
+  @Test def concatenated_list() {
+    testQuery(
+      title = "Using `UNWIND` with any expression returning a list",
+      text = "Any expression that returns a list may be used with `UNWIND`.",
+      queryText =
+        """WITH [1, 2] AS a, [3, 4] AS b
+           UNWIND (a + b) AS x
+           RETURN x""",
+      optionalResultExplanation = "The two lists -- _a_ and _b_ -- are concatenated to form a new list, which is then operated upon by `UNWIND`.",
+      assertions = (p) => assertEquals(List(1, 2, 3, 4), p.columnAs[Int]("x").toList)
+    )
+  }
+
+  @Test def nested_lists() {
+    testQuery(
+      title = "Using `UNWIND` with a list of lists",
+      text = "Multiple `UNWIND` clauses can be chained to unwind nested list elements.",
+      queryText =
+        """WITH [[1, 2], [3, 4], 5] AS nested
+          UNWIND nested AS x
+          UNWIND x AS y
+          RETURN y""",
+      optionalResultExplanation = "The first `UNWIND` results in three rows for `x`, each of which contains an element of the original list (two of which are also lists); namely, `[1, 2]`, `[3, 4]` and `5`. " +
+        "The second `UNWIND` then operates on each of these rows in turn, resulting in five rows for `y`.",
+      assertions = (p) => assertEquals(List(1, 2, 3, 4, 5), p.columnAs[Int]("y").toList)
+    )
+  }
+
+  @Test def empty_list() {
+    testQuery(
+      title = "Using `UNWIND` with an empty list",
+      text = "Using an empty list with `UNWIND` will produce no rows, irrespective of whether or not any rows existed beforehand, or whether or not other values are being projected. " +
+        "Essentially, `UNWIND []` reduces the number of rows to zero, and thus causes the query to cease its execution, returning no results.",
+      queryText =
+        """UNWIND [] AS empty
+           RETURN empty, 'literal_that_is_not_returned'""".stripMargin,
+      assertions = (p) => assertEquals(List(), p.columnAs[Int]("empty").toList)
+    )
+  }
+
+  @Test def using_non_lists() {
+    testQuery(
+      title = "Using `UNWIND` with an expression that is not a list",
+      text = "Attempting to use `UNWIND` on an expression that does not return a list will cause an error. " +
```

Comment:
Two concerns with this test:

* The test claims to test `using_non_list`, but in fact tests `unwind on null`, which is a related concept. I think perhaps two tests, one for null and one for a non-null, non-list.
* Is it really true that a non-list gives an error? I thought `UNWIND 5` would be the same as `UNWIND [5]`, and there is an example above in `nested_lists` that demonstrates that behaviour.

A third comment for the CLG is why null behaves differently to other values, when `UNWIND [null]` should return a single row containing null, and `UNWIND 5' should return a single row containing 5, why should `UNWIND null` not return a single row containing null?

Reply:
I have amended the description to include a mention of how something along the lines of `UNWIND 5` is an error. This is actually a bug, and the CLG have it on the wall (raised by Edinburgh as a result of the formal semantics work). When the bug is resolved - and how this will look remains to be decided - the docs will change accordingly